### PR TITLE
[SPARK-48340][PYTHON][FOLLOWUP] Support TimestampNTZ infer schema miss prefer_timestamp_ntz

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
Add UT

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Added UT

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
ping @HyukjinKwon

**Review by HyukjinKwon — APPROVED** [review]

File: python/pyspark/sql/tests/test_types.py [hunk 1]
```diff
@@ -236,14 +238,24 @@ def __init__(self):
...
         self.assertEqual(actual, expected)
 
         with self.sql_conf({"spark.sql.timestampType": "TIMESTAMP_NTZ"}):
             with self.sql_conf({"spark.sql.session.timeZone": "America/Sao_Paulo"}):
-                df = self.spark.createDataFrame([(datetime.datetime(1970, 1, 1, 0, 0),)])
+                data = [(
+                    datetime.datetime(1970, 1, 1, 0, 0),
+                    Row(a=Row(a=datetime.datetime(1970, 1, 1, 0, 0))),
+                )]
+                df = self.spark.createDataFrame(data)
```
Comment (HyukjinKwon) [inline python/pyspark/sql/tests/test_types.py:251]:
```suggestion
                data = [
                    (
                        datetime.datetime(1970, 1, 1, 0, 0),
                        Row(a=Row(a=datetime.datetime(1970, 1, 1, 0, 0))),
                    )
                ]
```

File: python/pyspark/sql/tests/test_types.py [hunk 2]
```diff
@@ -236,14 +238,24 @@ def __init__(self):
...
                 self.assertEqual(list(df.schema)[0].dataType.simpleString(), "timestamp_ntz")
                 self.assertEqual(df.first()[0], datetime.datetime(1970, 1, 1, 0, 0))
+                self.assertEqual(
+                    list(df.schema)[1].dataType.simpleString(),
+                    "struct<a:struct<a:timestamp_ntz>>"
```
Comment (HyukjinKwon) [inline python/pyspark/sql/tests/test_types.py:256]:
```suggestion
                    list(df.schema)[1].dataType.simpleString(), "struct<a:struct<a:timestamp_ntz>>"
```

File: python/pyspark/sql/tests/test_types.py [hunk 3]
```diff
@@ -236,14 +238,24 @@ def __init__(self):
...
 
         with self.sql_conf({"spark.sql.timestampType": "TIMESTAMP_NTZ"}):
             with self.sql_conf({"spark.sql.session.timeZone": "America/Sao_Paulo"}):
-                df = self.spark.createDataFrame([(datetime.datetime(1970, 1, 1, 0, 0),)])
+                data = [
+                    (
+                        datetime.datetime(1970, 1, 1, 0, 0),
+                        Row(a=Row(a=datetime.datetime(1970, 1, 1, 0, 0))),
+                    )
+                ]
```
Comment (HyukjinKwon) [inline python/pyspark/sql/tests/test_types.py:252]:
```suggestion
                ]
                df = self.spark.createDataFrame(data)
```

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**